source/tasks/BuildInformation/BuildInformationV6/buildInformation.ts

```diff
@@ -0,0 +1,189 @@
+    releaseId: string;
+    releaseUri: string;
+}
+
+export interface BuildEnvironmentVariables {
+    buildNumber: string;
+    buildId: number;
+    buildName: string;
+    buildRepositoryName: string;
+    buildRepositoryProvider: string;
+    buildRepositoryUri: string;
+    buildSourceVersion: string;
+}
+
+export interface AgentEnvironmentVariables {
+    agentBuildDirectory: string;
+}
+
+export interface SystemEnvironmentVariables {
+    projectName: string;
+    projectId: string;
+    teamCollectionUri: string;
+    defaultWorkingDirectory: string;
+}
+
+export type VstsEnvironmentVariables = ReleaseEnvironmentVariables & BuildEnvironmentVariables & AgentEnvironmentVariables & SystemEnvironmentVariables;
+
+export class BuildInformation {
+    constructor(readonly connection: OctoServerConnectionDetails, readonly logger: Logger, readonly task: TaskWrapper) {}
+
+    public async run() {
+        const environment = this.getVstsEnvironmentVariables();
+        const vstsConnection = this.createVstsConnection(environment);
+
+        const branch = await this.getBuildBranch(vstsConnection, environment);
+
+        const inputParameters = getInputParameters(parseInt(this.task.getVariable("system.jobAttempt") || "0") > 1, this.logger, this.task);
+
+        const config: ClientConfiguration = {
+            userAgentApp: "AzureDevOps (push-build-information;v6)",
+            instanceURL: this.connection.url,
+            apiKey: this.connection.apiKey,
+            logging: this.logger,
+        };
+        const client = await Client.create(config);
+
+        const commits = await this.getBuildChanges(vstsConnection, environment);
+
+        const packages: PackageIdentity[] = [];
+        for (const packageId of inputParameters.packages) {
+            packages.push({
+                Id: packageId,
+                Version: inputParameters.version,
+            });
+        }
+
+        const command: CreateOctopusBuildInformationCommand = {
+            spaceName: inputParameters.space,
+            BuildEnvironment: "Azure DevOps",
+            BuildNumber: environment.buildNumber,
+            BuildUrl: environment.teamCollectionUri.replace(/\/$/, "") + "/" + environment.projectName + "/_build/results?buildId=" + environment.buildId,
+            Branch: branch || "",
+            VcsType: this.getVcsTypeFromProvider(environment.buildRepositoryProvider),
+            VcsRoot: environment.buildRepositoryUri,
+            VcsCommitNumber: environment.buildSourceVersion,
+            Commits: commits.map((change) => ({ Id: change.id || "", Comment: change.message || "" })),
+            Packages: packages,
+        };
+
+        if (!environment.agentBuildDirectory) {
+            tasks.error("The Build Information step requires build information and therefore is not compatible with use in a Release pipeline.");
+            return;
+        }
+
+        const buildInformationDir = path.join(environment.agentBuildDirectory, "octo");
```

Comment:
We shouldn't need this code anymore, I wouldn't have thought. We don't have to write the Json down on disk any more, we should be just creating the structures in memory and handing them straight to the API client?